src/java/voldemort/store/rocksdb/RocksDbStorageConfiguration.java

```diff
@@ -34,13 +37,39 @@
 
     private Map<String, RocksDbStorageEngine> stores = new HashMap<String, RocksDbStorageEngine>();
 
+    // RocksDB Configuration Options
+    private final CompactionStyle compactionStyle;
+    private final CompressionType compressionType;
+    private final int maxBackgroundCompactions;
+    private final int maxBackgroundFlushes;
+    private final long maxBytesForLevelBase;
+    private final int maxWriteBufferNumber;
+    private final int statsDumpPeriodSec;
+    private final long targetFileSizeBase;
+    private final int targetFileSizeMultiplier;
+    private final long writeBufferSize;
+
     public RocksDbStorageConfiguration(VoldemortConfig config) {
         /**
          * - TODO 1. number of default locks need to debated. This default is
          * same as that of Krati's. 2. Later add the property to VoldemortConfig
          */
         this.voldemortconfig = config;
-        this.lockStripes = config.getAllProps().getInt("rocksdb.lock.stripes", 50);
+        Props props = config.getAllProps();
+
+        this.lockStripes = props.getInt("rocksdb.lock.stripes", 50);
+
+        // TODO: Validate the default mandatory options
+        compactionStyle = CompactionStyle.valueOf(props.getString("rocksdb.options.compactionStyle", CompactionStyle.UNIVERSAL.toString()));
```

Comment:
Hmm... it always brings a tear to my eyes to see inconsistencies in the code, but I guess that's a fight we'll take on another day (: ...

Going forward though, it'd be good to try to consolidate configs in ```VoldemortConfig``` as much as possible.

Thanks again for your work!

-F